scr/main/java/bittercode/util/DBUtil.java

```diff
@@ -0,0 +1,39 @@
+package com.bittercode.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.model.StoreException;
+
+public class DBUtil {
+
+    private static Connection connection;
+
+    static {
+
+        try {
+
+            Class.forName(DatabaseConfig.DRIVER_NAME);
+            
+            connection = DriverManager.getConnection(DatabaseConfig.CONNECTION_STRING, DatabaseConfig.DB_USER_NAME,
+                    DatabaseConfig.DB_PASSWORD);
+        } catch (SQLException | ClassNotFoundException e) {
+
+            e.printStackTrace();
+
+        }
+
+    }// End of static block
+
+    public static Connection getConnection() throws StoreException {
+
+        if (connection == null) {
+            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE);
+        }
+
+        return null;

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Incorrect return value in getConnection method</b></div>
<div id="fix">

The `getConnection()` method is returning `null` instead of the `connection` object. This could lead to `NullPointerException`s when the method is called. Consider returning the `connection` object instead.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -36,1 +36,1 @@
 -        return null;
 +        return connection;
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/5#issuecomment-2435141219>#98533c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged